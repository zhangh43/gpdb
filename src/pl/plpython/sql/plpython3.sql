create language plpython3u;

CREATE FUNCTION import_succeed() returns text AS $$
  import sys
  import numpy
  return "succeeded, as expected"
$$ LANGUAGE plpython3u;

SELECT import_succeed();

CREATE TYPE named_value AS (
	  name  text,
	  value  integer);

CREATE OR REPLACE FUNCTION make_pair_sets (name text)
RETURNS SETOF named_value AS $$
  import numpy as np
  return ((name, i) for i in np.arange(1))
$$ LANGUAGE plpython3u;

SELECT * from make_pair_sets('test');
