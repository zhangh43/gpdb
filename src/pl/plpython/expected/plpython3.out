create language plpython3u;
CREATE FUNCTION import_succeed3() returns text AS $$
  import sys
  import numpy
  return "succeeded, as expected"
$$ LANGUAGE plpython3u;
SELECT import_succeed3();
    import_succeed3     
------------------------
 succeeded, as expected
(1 row)

CREATE TYPE named_value3 AS (
	  name  text,
	  value  integer);
CREATE OR REPLACE FUNCTION make_pair_sets3 (name text)
RETURNS SETOF named_value3 AS $$
  import numpy as np
  return ((name, i) for i in np.arange(1))
$$ LANGUAGE plpython3u;
SELECT * from make_pair_sets3('test');
 name | value 
------+-------
 test |     0
(1 row)

